waitress/parser.py

```diff
@@ -291,8 +291,12 @@ def crack_first_line(line):
             version = m.group(5)
         else:
             version = None
-        command = m.group(1).upper()
+        method = m.group(1)
+
+        # the
+        if method != method.upper():
+            raise ParsingError('Malformed HTTP method "%s"' % tostr(method))
```

Comment:
Is there a reason to force it to be upper specifically instead of just passing it through untouched?